include/s3select.h

```diff
@@ -2238,10 +2250,49 @@ class base_s3object
       m_where_clause->traverse_and_apply(m_sa, m_s3_select->get_aliases(), m_s3_select->is_json_query());
     }
 
+    for (auto& a : *m_s3_select->get_aliases()->get())
+    {
+        m_alias_keys.push_back(a.first);
+        set.insert(a.second);
+    }
+    
+    size_t m_alias_index = 0;
+    int index_json_projection = 0;
+    is_json = m_s3_select->is_json_query();
+
     for (auto& p : m_projections)
     {
+      if(p->is_statement_contain_star_operation())
+      {
+        set_star_true();
+      }
       p->traverse_and_apply(m_sa, m_s3_select->get_aliases(), m_s3_select->is_json_query());
+
+      std::string key_from_projection{};
+      if(p->is_column()){
+        key_from_projection = p->get_key_from_projection();
+      }
+
+      if(set.count(p) == 0 && p->is_column())  {
+        m_projection_keys.push_back(key_from_projection);
+      } else if(set.count(p) > 0 && p->is_column()) {
+          m_projection_keys.push_back(m_alias_keys[m_alias_index++]);
+      } else if(!p->is_column() && m_csv_defintion.output_json_format && set.count(p) > 0 )  {
```

Comment:
once these conditions will reside on a separate code-scope, it's possible to simplify this code. 
remove the `m_csv_defintion.output_json_format`, and reduce the number of conditions.